epochStart/bootstrap/epochStartMetaBlockProcessor.go

```diff
@@ -191,6 +192,27 @@ func (e *epochStartMetaBlockProcessor) GetEpochStartMetaBlock(ctx context.Contex
 	}
 }
 
+func (e *epochStartMetaBlockProcessor) getMostReceivedMetaBlock() (*block.MetaBlock, error) {
+	e.mutReceivedMetaBlocks.RLock()
+	defer e.mutReceivedMetaBlocks.RUnlock()
+
+	const hashNotAvailable = "N/A"
+	mostReceivedHash := hashNotAvailable
+	maxLength := minNumOfPeersToConsiderBlockValid - 1
+	for hash, entry := range e.mapMetaBlocksFromPeers {
+		if len(entry) > maxLength {
+			maxLength = len(entry)
+			mostReceivedHash = hash
+		}
+	}
+
+	if mostReceivedHash == hashNotAvailable {
+		return nil, epochStart.ErrTimeoutWaitingForMetaBlock
+	}
+
+	return e.mapReceivedMetaBlocks[mostReceivedHash], nil
```

Comment:
this returns the first miniblock of a specific length. Is it ok if multiple blocks have the same number of appearances to return the first? e.g.  we have 20 peers but 2 blocks have 9 appearances.

Reply:
You're right, this can happen. I don't know what to say about this.. maybe @sasurobert can provide an input

Reply:
this is only to resolve an edge case where you do not get the same block from 2/3+1 of the connected peers. In that case the best thing you can do is to choose the best of the rest. If it is an equality - you can't decide which is better. It should not happen. Anything you decide in that moment is ok. at least you decided.

Reply:
ok